### PR TITLE
[prometheus] Change grafana version in `patches/build_go.patch.tpl` automatically from docker arguments

### PR DESCRIPTION
**Pull request by name212**
Signed-off-by: Nikolay Mitrofanov <nikolay.mitrofanov@flant.com>

## Description
Add grafana version from docker args to  `patches/build_go_patch` automatically via `sed`

## Why do we need it, and what problem does it solve?
In release 1.34 we changed grafana version in docker file but forget change version in the patch and grafana show incorrect version on home dashboard and with `-v` argument.
Now it will change automatically.

## What is the expected result?
We should see `8.5.9` version on home dashboard.

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: prometheus
type: fix
summary: Change grafana version in `patches/build_go.patch.tpl` automatically from docker arguments
impact_level: default
```

### CODE REVIEW

File: modules/300-prometheus/images/grafana/patches/build_go.patch.tpl [hunk 1]
```diff
@@ -8,7 +8,7 @@ index bd0128d..b4d7550 100644
  	b.WriteString("-w")
 -	b.WriteString(fmt.Sprintf(" -X main.version=%s", opts.version))
 -	b.WriteString(fmt.Sprintf(" -X main.commit=%s", getGitSha()))
-+	b.WriteString(fmt.Sprintf(" -X main.version=%s", "8.5.2"))
++	b.WriteString(fmt.Sprintf(" -X main.version=%s", "GRAFANA_VERSION_HERE"))
```
Comment (nabokihms) [inline modules/300-prometheus/images/grafana/patches/build_go.patch.tpl:11]:
```suggestion
+	b.WriteString(fmt.Sprintf(" -X main.version=%s", os.Getenv("GRAFANA_VERSION_HERE")))
```

File: modules/300-prometheus/images/grafana/Dockerfile [hunk 2]
```diff
@@ -27,8 +27,8 @@ RUN patch -p1 < ./feat_prometheus_extra_vars.patch
 COPY ./patches/fix_heatmap_thin_bars_on_ff.patch .
 RUN patch -p1 < ./fix_heatmap_thin_bars_on_ff.patch
 # Set more useful version than 'dev'. There are tabs in patch, so -l is used.
-COPY ./patches/build_go.patch .
-RUN patch -p1 -l < ./build_go.patch
+COPY patches/build_go.patch.tpl .
+RUN sed "s/GRAFANA_VERSION_HERE/${GRAFANA_VERSION}/g" ./build_go.patch.tpl > ./build_go.patch && patch -p1 -l < ./build_go.patch
```
Comment (nabokihms) [inline modules/300-prometheus/images/grafana/Dockerfile:31]:
```suggestion
RUN patch -p1 -l < ./build_go.patch
ENV GRAFANA_VERSION=${GRAFANA_VERSION}
```

**Review by nabokihms — CHANGES_REQUESTED** [review]

**Comment by name212** [comment]
@nabokihms your suggestions do not work :-(

```
Defaulted container "grafana" out of: grafana, dashboard-provisioner, kube-rbac-proxy
nobody@grafana-95d64d6dd-fpw5n:/usr/share/grafana$ grafana-server -v
Version  (commit: fix_heatmap,feat_extra_vars, branch: main)
```

**Comment by name212** [comment]
fixed

**Review by nabokihms — APPROVED** [review]

**Pull request merged**